packages/rpc-core/src/rpc-methods/index.ts

```diff
@@ -113,11 +113,25 @@ export type SolanaRpcMethods = GetAccountInfoApi &
     SendTransactionApi &
     SimulateTransactionApi;
 
-export function createSolanaRpcApi(config?: Config): IRpcApi<SolanaRpcMethods> {
-    return createJsonRpcApi<SolanaRpcMethods>({
+export function createSolanaRpcApi<TCluster extends 'devnet' | 'testnet' | 'mainnet' | void = void>(
+    config?: Config,
+): TCluster extends 'devnet'
+    ? IRpcApiDevnet<SolanaRpcMethods>
+    : TCluster extends 'testnet'
+      ? IRpcApiTestnet<SolanaRpcMethods>
+      : TCluster extends 'mainnet'
+        ? IRpcApiMainnet<Omit<SolanaRpcMethods, 'requestAirdrop'>>
+        : IRpcApi<SolanaRpcMethods> {
+    return createJsonRpcApi<SolanaRpcMethods, TCluster>({
         parametersTransformer: getParamsPatcherForSolanaLabsRpc(config) as (params: unknown[]) => unknown[],
         responseTransformer: patchResponseForSolanaLabsRpc,
-    });
+    }) as TCluster extends 'devnet'
+        ? IRpcApiDevnet<SolanaRpcMethods>
+        : TCluster extends 'testnet'
+          ? IRpcApiTestnet<SolanaRpcMethods>
+          : TCluster extends 'mainnet'
+            ? IRpcApiMainnet<Omit<SolanaRpcMethods, 'requestAirdrop'>>
+            : IRpcApi<SolanaRpcMethods>;
```

Comment:
This type condition is used a lot throughout your stack and it might be helpful to have it extracted as a type helper like `ISolanaRpcApiFromCluster<TCluster>`

Reply:
Hah, didn't even think of that tbh. I'll add!